src/compile.ts

```diff
@@ -72,18 +72,59 @@ export function compile(
       );
 
       function visitor(node: ts.Node): ts.Node {
-        const _visit = () => {
-          if (isAppsyncResolver(node)) {
-            return visitAppsyncResolver(node);
-          } else if (isReflectFunction(node)) {
-            return errorBoundary(() =>
-              toFunction("FunctionDecl", node.arguments[0])
-            );
-          }
-          return node;
-        };
+        if (isAppsyncResolver(node)) {
+          return visitAppsyncResolver(node as ts.NewExpression);
+        } else if (isStepFunction(node)) {
+          return visitStepFunction(node as ts.NewExpression);
+        } else if (isReflectFunction(node)) {
+          return errorBoundary(() =>
+            toFunction("FunctionDecl", node.arguments[0])
+          );
+        }
         // keep processing the children of the updated node.
-        return ts.visitEachChild(_visit(), visitor, ctx);
+        return ts.visitEachChild(node, visitor, ctx);
+      }
```

Comment:
regression here, (apparently the tests for this are  on my change).

```ts
   bus
      .when(...)
     .map(...)
```

the `map` will be visited, but the `when` will not as its just a child of `map`.

Reply:
should we be transforming nodes we create? Why doesn't the transformer recurse?

Reply:
Richer response in #39 . 

https://github.com/sam-goodwin/functionless/pull/39/files/5087c05d71f2d079e5a40c013cbbb69b1150a1ee#r838371988

Basically we stop the recursion to transform the current node and the children (ex: the object that the current method is called from) is never visited.

Only a problem once we have chained methods like event bridge uses.

```
bus.when().map()
```

If `.map()` is caught and transformed, `bus.when()` is never processed.

Reply:
Reverted.